tests/components/tradfri/test_init.py

```diff
@@ -68,3 +75,45 @@ async def test_config_json_host_imported(hass, mock_gateway_info, mock_entry_set
     assert config_entry.domain == "tradfri"
     assert config_entry.source == "import"
     assert config_entry.title == "mock-host"
+
+
+async def test_entry_setup_unload(hass, api_factory, gateway_id):
+    """Test config entry setup and unload."""
+    entry = MockConfigEntry(
+        domain=tradfri.DOMAIN,
+        data={
+            tradfri.CONF_HOST: "mock-host",
+            tradfri.CONF_IDENTITY: "mock-identity",
+            tradfri.CONF_KEY: "mock-key",
+            tradfri.CONF_IMPORT_GROUPS: True,
+            tradfri.CONF_GATEWAY_ID: gateway_id,
+        },
+    )
+
+    entry.add_to_hass(hass)
+    with patch.object(
+        hass.config_entries, "async_forward_entry_setup", return_value=True
+    ) as setup:
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+        assert setup.call_count == len(tradfri.PLATFORMS)
+
+    dev_reg = await async_get_device_registry(hass)
+    dev_entries = async_entries_for_config_entry(dev_reg, entry.entry_id)
+
+    assert dev_entries
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6103615)